pkg/reposync/conditions.go

```diff
@@ -110,9 +111,8 @@ func SetSyncing(rs *v1beta1.RepoSync, status bool, reason, message, commit strin
 }
 
 // setCondition adds or updates the specified condition with a True status.
-// It returns a boolean indicating if the condition status is transited.
-func setCondition(rs *v1beta1.RepoSync, condType v1beta1.RepoSyncConditionType, status metav1.ConditionStatus, reason, message, commit string, errorSources []v1beta1.ErrorSource, errorSummary *v1beta1.ErrorSummary, lastUpdate metav1.Time) bool {
-	conditionTransited := false
+// It returns a boolean indicating if the condition status has transitioned.
+func setCondition(rs *v1beta1.RepoSync, condType v1beta1.RepoSyncConditionType, status metav1.ConditionStatus, reason, message, commit string, errorSources []v1beta1.ErrorSource, errorSummary *v1beta1.ErrorSummary, lastUpdate metav1.Time) (updated, transitioned bool) {
```

Comment:
Is it intentional that the `updated` return value is unused by the consumers of this function?

Reply:
The point of the updated return was to allow the caller not to have to do its own DeepEquals to check if anything changed, but I didn't want to get too distracted doing that refactor, since this was just something I was quickly fixing while doing the finalizer work. And it sounds like @xinnywinne wants to change the updates in the reconciler-manager to use SSA anyway, so she's going to be changing this code soon. When doing an SSA we can't really do a full DeepEquals on the RSync, because there might be other fields changed that we don't control. So having this update boolean might allow us to skip some SSA calls, if we know the condition hasn't changed.

Reply:
I routed the updates through the public Set calls so they can now be used by the reconciler and reonciler-manager, when we decide to do that refactor.